src/GameOptions/ui/KeyBindingPage.tsx

```diff
@@ -0,0 +1,308 @@
+import { Modal } from "../../ui/React/Modal";
+import { SimplePage } from "../../ui/Router";
+import { KEYCODE } from "../../utils/helpers/keyCodes";
+import {
+  convertKeyboardEventToKeyCombination,
+  defaultKeyBinding,
+  determineKeyBindingTypes,
+  getKeyCombination,
+  isKeyCombinationPressed,
+  isSpoilerKeyBindingType,
+  KeyBindingEvents,
+  KeyBindingEventType,
+  parseKeyCombinationToString,
+  SpoilerKeyBindingTypes,
+  type KeyBindingType,
+  type KeyCombination,
+} from "../../utils/KeyBindingUtils";
+import { GameOptionsPage } from "./GameOptionsPage";
+import { dialogBoxCreate } from "../../ui/React/DialogBox";
+import { knowAboutBitverse } from "../../BitNode/BitNodeUtils";
+
+function determineConflictKeys(
+  keyBindingType: KeyBindingType,
+  isPrimary: boolean,
+  newCombination: KeyCombination,
+): Set<string> {
+  const conflicts: Set<string> = determineKeyBindingTypes(Settings.KeyBindings, newCombination);
+  // Check if the new combination is the same as the current key binding.
+  if (conflicts.has(keyBindingType)) {
+    const currentKeyBinding = getKeyCombination(Settings.KeyBindings, keyBindingType, isPrimary);
+    if (
+      currentKeyBinding &&
+      currentKeyBinding.control === newCombination.control &&
+      currentKeyBinding.alt === newCombination.alt &&
+      currentKeyBinding.shift === newCombination.shift &&
+      currentKeyBinding.meta === newCombination.meta &&
+      currentKeyBinding.code === newCombination.code
+    ) {
+      conflicts.delete(keyBindingType);
+    }
+  }
+  // Common single-key hotkeys.
+  if (
+    isKeyCombinationPressed(newCombination, { code: KEYCODE.ESC }) ||
+    isKeyCombinationPressed(newCombination, { code: KEYCODE.ENTER }) ||
+    isKeyCombinationPressed(newCombination, { code: KEYCODE.NUMPAD_ENTER }) ||
+    isKeyCombinationPressed(newCombination, { code: KEYCODE.TAB })
+  ) {
+    conflicts.add("Common hotkeys");
+  }
+  // Copy - Paste - Cut
+  if (window.navigator.userAgent.includes("Mac")) {
+    if (
+      isKeyCombinationPressed(newCombination, { meta: true, code: KEYCODE.C }) ||
+      isKeyCombinationPressed(newCombination, { meta: true, code: KEYCODE.V }) ||
+      isKeyCombinationPressed(newCombination, { meta: true, code: KEYCODE.X })
+    ) {
+      conflicts.add("Common hotkeys");
+    }
+  } else {
+    if (
+      isKeyCombinationPressed(newCombination, { control: true, code: KEYCODE.C }) ||
+      isKeyCombinationPressed(newCombination, { control: true, code: KEYCODE.V }) ||
+      isKeyCombinationPressed(newCombination, { control: true, code: KEYCODE.X })
+    ) {
+      conflicts.add("Common hotkeys");
+    }
+  }
+  // Terminal-ClearScreen
+  if (isKeyCombinationPressed(newCombination, { control: true, code: KEYCODE.L })) {
+    conflicts.add("Terminal-ClearScreen");
+  }
+  // Bash hotkeys
+  if (
+    Settings.EnableBashHotkeys &&
+    (isKeyCombinationPressed(newCombination, { control: true, code: KEYCODE.M }) ||
```

Comment:
The bash hotkeys *definitely* should be bindable, but that can also be for another day.